File: src/main/java/com/wordonline/matching/matching/dto/SessionDto.java
package com.wordonline.matching.matching.dto;

public record SessionDto(
        String sessionId,
        Long uid1,
        Long uid2,
        SessionType sessionType,
        Long scenarioId
) {

    public static SessionDto PVP(String sessionId, Long uid1, Long uid2) {
        return new SessionDto(sessionId, uid1, uid2, SessionType.PVP, null);
    }

    public static SessionDto PVE(String sessionId, Long userId, Long scenarioId) {
        return new SessionDto(sessionId, userId, null, SessionType.PVE, scenarioId);
    }

    public static SessionDto Practice(String sessionId, Long uid1, Long uid2) {
        return new SessionDto(sessionId, uid1, uid2, SessionType.Practice, null);
    }

    public static SessionDto from(String sessionId, long uid1, long uid2) {
        if (uid2 < 0) {
            return SessionDto.Practice(sessionId, uid1, uid2);
        } else {
            return SessionDto.PVP(sessionId, uid1, uid2);
        }
    }
}

File: src/main/kotlin/com/wordonline/matching/matching/dto/SessionDto.kt
package com.wordonline.matching.matching.dto

data class SessionDto(
    val sessionId: String,
    val uid1: Long,
    val uid2: Long?,
    val sessionType: SessionType,
    val scenarioId: Long?,
) {
    companion object {
        fun PVP(sessionId: String, uid1: Long, uid2: Long) =
            SessionDto(sessionId, uid1, uid2, SessionType.PVP, null)

        fun PVE(sessionId: String, userId: Long, scenarioId: Long) =
            SessionDto(sessionId, userId, null, SessionType.PVE, scenarioId)

        fun Practice(sessionId: String, uid1: Long, uid2: Long) =
            SessionDto(sessionId, uid1, uid2, SessionType.Practice, null)

        fun from(sessionId: String, uid1: Long, uid2: Long) =
            if (uid2 < 0) Practice(sessionId, uid1, uid2) else PVP(sessionId, uid1, uid2)
    }
}
